atlas/packages/app/src/components/VideoGallery.tsx

```diff
@@ -0,0 +1,127 @@
+		showChannel: true,
+		views: "345k",
+		time: "2 weeks ago",
+		poster:
+			"https://s3-alpha-sig.figma.com/img/4baa/8670/e77491e871584e97b955d728627020f3?Expires=1593388800&Signature=LkSz8F8wftV81dxfumpYE5CToghiq5a3hvwywrhwVAXFwQvNzfymCImBCY9wIxILX4nnOlaNX0g~~iZ2RXD104VvLn6q967Hlg0uAH1h3eutIZzb4sQ0i4R-wXUw5ZNy2QjLCOIlKuCXMXSO1UiI8t8CJtdel9XaSqm1u~JsPFhRjRlO60RCi2cM2pJ~8G~kofCMT6J3NMgyRruVLUDY9FPGh5dI5XRFmeKbMThO2IT5KfUGsPDLb98i8gBrkLe2EKMx2DxHlDxcA3j2LVtf4yaGG-EmeDrQflknjdoAY0BKfC8Mob10ldGsYnJsMHOraz971POWOVpWTuoF0h6nNg__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
+	},
+	{
+		title: "Sample Video Title",
+		channel: "Channel Name",
+		showChannel: true,
+		views: "345k",
+		time: "2 weeks ago",
+		poster:
+			"https://s3-alpha-sig.figma.com/img/e0d6/4163/147cd1502ba0a99ea7258f31be9812b3?Expires=1593388800&Signature=av5e4oaaYXVDxA9U0DAQKuwA-vyC4VRaGYz2lO5IIaz-CP55B717sXn-V6H6JzMNZ7JavA9q8~uMK-r9jd-yxoia1UopEiUeSkZgQlbBhd4DEVBPoMhMF3nC--mZ4bQnVvK9xN933TCm9z46jkAOZ7U8UYp5TlA829Rnmgw3QkaNlPLW5UkgU04k6pAae5Y3t9pAnwErmDCDjHuaXKrCjCuG0zwk84f0JQAcZt4VojD8kVDFCf0IOHdZtIQMUk47CQfdWJ2KDBkm1oOcVAweTSHXdwzpiFQ~o1i-hZVYtgzIiGTE6IqW3qsksqiLtWwMsui2dvWXVu2actRRwR~OTg__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
+	},
+	{
+		title: "Sample Video Title",
+		channel: "Channel Name",
+		showChannel: true,
+		views: "345k",
+		time: "2 weeks ago",
+		poster:
+			"https://s3-alpha-sig.figma.com/img/63ab/b514/2a10cf17ceaf00972e7c12c18cacd832?Expires=1593388800&Signature=WRk44Vxkc1mYbXSdGroHP8Hvd4dfhtWMukS~2yfNlY6KSb94ZzeFaKID8dMj4wznZo-sSp~lAagV7NJMFO9QLCEw0E7YeX3TnOWeFDrx63wkc388h-srSxxrNVUpB6WUCfkPtdusDrfKOx2HvM6zZLlSgYs4-McX2VTJ-g7bKx2Pzt64mox6wTGlbXWREkV2sbLSBSF6P0fpwge3VPt5jX1E~vDJ8X4mtZgN1DRIQMtc2dfZhZhyL1jNeIZe5WJyOcY9b2E3KL54pWt13UGsI08nTzloqOPFAt~fHwbwUbwWeelbjkzSABh55KYVNePerISUIy~esGJ11px7l9oWTw__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
+	},
+	{
+		title: "Sample Video Title",
+		channel: "Channel Name",
+		showChannel: true,
+		views: "345k",
+		time: "2 weeks ago",
+		poster:
+			"https://s3-alpha-sig.figma.com/img/b327/39be/143853e8ca345927ee54a7ca9acaed0c?Expires=1593388800&Signature=G1vikiLUBwdBdrzS3RC1WScSJSAAD7FEteNX4bfPf0VQL0w9fsMfha31S-l6EHxU~WTh3GAic8X5furwHjwLbE2nVUIiA0q6Rizxfd8kT97Z-9uQzPbCkla6a1qIX-7wq19t2evAVST--Sz7wfEYVCUTdZaEmqgBsJnYI3fl7nzrwLhaaTTN3liOTltgvlQn8btRClliTWENyHBwu6cwvMHuX6ePt5swhIj4gz34F0MyY~PtnsFbBHFerShd3t2DYsMqDv8CEB089v6IYZh87Rnqk7jJFZC2ZBULhjlwKQpboJOI35cov3aGOnuTe5vcmULLSfEXuV5kWJrtzJnQlA__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
+	},
+];
+
+const articleStyles = css`
+	max-width: 320px;
+	margin: auto ${theme.spacing.m};
+`;
+
+export default function VideoGallery({ title = "" }: VideoGalleryProps) {
+	const videos = videoPlaceholders.concat(videoPlaceholders).concat(videoPlaceholders);
+	const [controlsTop, setControlsTop] = useState<SerializedStyles>(css``);
+	const imgRef = useCallback((node: HTMLImageElement) => {
+		if (node != null) {
+			setControlsTop(css`
+				top: calc(${Math.round(node.clientHeight) / 2}px - 24px);
```

Comment:
Could you explain what's the purpose of this? (passing ref, etc.)

Reply:
It seems like we want to get the height of the image for some reason, but that height is directly controlled by the prop of `VideoPreview`. That tells me there's some issue here. It seems to me we should either:
a) set the height via prop and not check the ref
b) have `VideoPreview` choose the height automatically and then check it via ref
Currently, in this component we're checking a value that's ours to set.

Another question is whether we should use `height` and `width` props at all. By using them we're leaking styling inside the code. Maybe `VideoPreview` should just have some default value that can be overridden via styling, not props?

Reply:
I needed to align the wheels of the carousel to the middle of the `VideoPreview` thumbnail / image but, in general, I had no way of knowing the height of a thumbnail at priori (things could be resized etc...). 
So my solution was to attach a callback ref to the image inside `VideoPreview` to get the height and set the correct position of carousel wheels. 

It feels a bit hacky, so I'm _very_ open to suggestions or different implementations.

Reply:
Does my 2nd answer give you some ideas? I think we may have posted at the same time

Reply:
Also, if the line is left as it is, we shouldn't use a magic value in the code that's not described in any way. `24px` here is just an arbitrary value and I have no idea what it's supposed to mean. Does it correlate to some other style? Does it have to be changed along with some other value in another file?

If possible in a case like this, this should be some imported variable, if not, then we should add a comment.

Reply:
Btw, why do we even need to pass the height to carousel controls? Shouldn't they just reside inside the carousel container and center themselves vertically automatically?

Reply:
Let me try to dissect what's going on here:

- `VideoGallery` is the common ancestor of both `Gallery` and each `VideoPreview` children.

- `Gallery` centers the carousel wheels in the middle of the container by default, there are however some cases, like when we want to center the wheels in the middle of a video thumbnail, where the center of the container is not the desired position of the wheels.
To account for those cases, `Gallery` exposes a `leftControlCss` and a `rightControlCss` prop, those are escape hatches to reposition the wheels from the center of the `Gallery` container. 

- We want to center the wheels in the middle of the video thumbnail, to do so we need to measure the height of the thumbnail (the purpose of the `imgRef`) and then reposition the Carousel wheels accordingly, this has to happen in `VideoGallery` since it is the common ancestor.

Now coming to your concerns:

- I don't understand what you mean by `width` and `height` props, if you refer to the ones of the thumbnail, so just hardcode them and get rid of the ref to clean code, I don't think that would work since I would expect `VideoPreview` to adjust it's own `width` and `height` to preserve a fixed aspect ratio on resizing, so an "external" component would have no way of knowing those values.

- The 24px is just the height of the Carousel wheel. I could either add a ref to measure it or label it with a value, I would just go for the labeling since it's the wheels don't resize on smaller screensizes.

Reply:
Okay, now I see what's going on with the manual styling of the wheels and that looks okay.

Regarding my concerns:
- `width` and `height` props: what I mean here is that `VideoPreview` exposes props to set these values. So `VideoGallery` as `VideoPreview` consumer is responsible for setting these values if it wants them other than the default. But even though `VideoGallery` is the de factor controller of `VideoPreview`'s cover `height`, it still checks it via a ref. So we have a communication issue here - we check a value that's ours to set. The easiest approach would probably be to remove `height` and `width` props from `VideoPreview`

- the `24px` value. Sure, if this will be a constant then there's no point of measuring it. My point was that we shouldn't use a value like this outside of its context without any description. If this is the height of the wheel, let's declare it alongside wheel styles and import here.

Reply:
@Gamaranto I believe this is still unresolved

Reply:
I see what you mean now, and I do agree with the points you raised. 
I can't see a trivial way to fix the first point, so I've just removed `width` and `height` from the props as you suggested. Regarding point 2, I've added it as a constant in `VideoGallery` since I believe height of the control whell might be contextual and adding it there naturally evolves towards measuring it with a ref if we ever need to.

Reply:
Awesome :)